src/main/java/org/grouplens/grapht/reflect/Satisfaction.java

```diff
@@ -114,5 +115,5 @@
      *         satisfaction, instantiated using the specified dependency
      *         mapping.
      */
-    Instantiator makeInstantiator(Map<Desire,Instantiator> dependencies);
+    Instantiator makeInstantiator(Map<Desire,Instantiator> dependencies, InjectionContainer injectionContainer);
```

Comment:
I would prefer to call the new parameter just `container` - excessively long variable names get cumbersome fast. Please make this change across all the instantiator classes.